### PR TITLE
analytics(screens): add tracking in profile and fix track event not being a `screen` event.

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Enhancement**

This PR resolves [CX-1449], [CX-1458] and one with no jira ticket, about profile screen tracking missing.

### Description

We found that Profile didn't send a screen tracking event. Now it does.

profile
Screenshot 2021-05-12 at 18 12 11

before, track:screen (highlighted)
Screenshot 2021-05-12 at 18 41 03

after, screen:screen (highlighted)
Screenshot 2021-05-12 at 18 42 01

extra
Screenshot 2021-05-12 at 18 33 04

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[CX-1449]: https://artsyproduct.atlassian.net/browse/CX-1449

[CX-1458]: https://artsyproduct.atlassian.net/browse/CX-1458

### CODE REVIEW

File: src/lib/utils/track/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { ActionType, Screen } from "@artsy/cohesion"
+
+export const screen = (info: Omit<Screen, "action">): Screen => ({
```
Comment (pvinis) [inline src/lib/utils/track/helpers.ts:3]: just a little helper for cohesion stuff

File: src/lib/utils/track/providers.tsx [hunk 2]
```diff
@@ -1,25 +1,19 @@
-import { ActionType } from "@artsy/cohesion"
+import { ActionType, Screen } from "@artsy/cohesion"
 
 import * as Schema from "./schema"
 
 interface CohesionAction {
   // TODO: This can be removed once cohesion provides a global `Action` type.
   action: string
 }
-interface CohesionScreen {
```
Comment (pvinis) [inline src/lib/utils/track/providers.tsx:9]: using the actual thing coming from cohesion

File: Artsy/App/ARAppDelegate+Emission.m [hunk 3]
```diff
@@ -169,13 +169,18 @@ - (void)setupSharedEmissionWithPackagerURL:(NSURL *)packagerURL;
...
         NSMutableDictionary *properties = [info mutableCopy];
-        if (info[@"action_type"] ) {
+        if (info[@"action_type"]) {
             // Track event
             [properties removeObjectForKey:@"action_type"];
             [ARAnalytics event:info[@"action_type"] withProperties:[properties copy]];
         } else if (info[@"action"]) {
-            // Track event
-            [ARAnalytics event:info[@"action"] withProperties:[properties copy]];
+            if ([info[@"action"] isEqualToString:@"screen"]) {
```
Comment (pvinis) [inline Artsy/App/ARAppDelegate+Emission.m:177]: here is the main change. the logic was different from android. now they are the same. if was not taking into account the cohesion way for `Screen`.

**Comment by pvinis** [comment]
When this is merged, we should make a beta build for ios and android. I'll tag @mikehrom and @abhitip so they get notifs for updates on this PR.

File: src/lib/utils/track/helpers.ts (hunk 1 above)
Comment (MounirDhahri) [reply src/lib/utils/track/helpers.ts:3]: good one👌

File: Artsy/App/ARAppDelegate+Emission.m (hunk 3 above)
Comment (MounirDhahri) [reply Artsy/App/ARAppDelegate+Emission.m:177]: **thought:** shouldn't we instead change android to match ios since this has been this way for a while?

**Review by ds300 — APPROVED** [review]

File: Artsy/App/ARAppDelegate+Emission.m (hunk 3 above)
Comment (MounirDhahri) [reply Artsy/App/ARAppDelegate+Emission.m:177]: great, thanks for explaining 🙌

**Comment by pvinis** [comment]
I'll make a beta for both platforms so data can test easily. Thanks for the reviews!

**Pull request merged**